common-app-lib/src/cmr/common_app/api/launchpad_token_validation.clj

```diff
@@ -1,17 +1,66 @@
 (ns cmr.common-app.api.launchpad-token-validation
   "Validate Launchpad token."
   (:require
+   [cheshire.core :as json]
+   [clojure.data.codec.base64 :as base64]
+   [clojure.string :as string]
    [cmr.common-app.config :as config]
    [cmr.common.services.errors :as errors]
    [cmr.transmit.config :as transmit-config]))
 
+;; Note: Similar code exists at gov.nasa.echo.kernel.service.authentication
 (def URS_TOKEN_MAX_LENGTH 100)
 
+(defn is-heritage-token?
+  "Heritage token, meaning not the original style legacy token, but a modern
+   token made to be identifiable but also comply with the legacy standard. These
+   tokens are prefixed with 'EDL+'.
+   Note: Similar code exists at gov.nasa.echo.kernel.service.authentication."
+  [token]
+  (string/starts-with? token "EDL+"))
+
+(defn is-legacy-token?
+  "A check of last resort, only used if all other checks have not passed.
```

Comment:
"A check of last resort"

If this is the last resort, then why does `is-launchpad-token?` check for `is-legacy-token?` 

I assume launchpad tokens are too large for one to ever get accidentally flagged as `false` from this check, but this docstring does seem to contradict the functionality of `is-launchpad-token?`

Reply:
no, that is not descriptive enough, when I had the tests in a different order, that comment was true, but the code is changed.